lib/quicly.c

```diff
@@ -900,15 +900,18 @@ static void update_idle_timeout(quicly_conn_t *conn, int is_in_receive)
     if (idle_msec == INT64_MAX)
         return;
 
-    uint32_t three_pto = 3 * quicly_rtt_get_pto(&conn->egress.loss.rtt, conn->super.ctx->transport_params.max_ack_delay);
+    uint32_t three_pto = 3 * quicly_rtt_get_pto(&conn->egress.loss.rtt, conn->super.ctx->transport_params.max_ack_delay,
+                                                conn->egress.loss.conf->min_pto);
     conn->idle_timeout.at = now + (idle_msec > three_pto ? idle_msec : three_pto);
     conn->idle_timeout.should_rearm_on_send = is_in_receive;
 }
 
 static void update_loss_alarm(quicly_conn_t *conn)
 {
-    quicly_loss_update_alarm(&conn->egress.loss, now, conn->egress.last_retransmittable_sent_at,
-                             conn->egress.sentmap.bytes_in_flight != 0 || conn->super.peer.address_validation.send_probe);
+    quicly_loss_update_alarm(
+        &conn->egress.loss, now, conn->egress.last_retransmittable_sent_at,
+        conn->egress.sentmap.bytes_in_flight != 0 || conn->super.peer.address_validation.send_probe,
+        conn->super.ctx->stream_scheduler->can_send(conn->super.ctx->stream_scheduler, conn, /*including_new_data=*/0));
```

Comment:
I think that `including_new_data` should be set to 1.

We want to arm the aggressive timer only when there's nothing to send. Either having new data or data to "re"transmit is not the case.